src/osha/oira/services/upload.py

```diff
@@ -0,0 +1,32 @@
+from lxml.etree import XMLSyntaxError
+from plone import api
+from plone.restapi.deserializer import json_body
+from plone.restapi.services import Service
+
+import base64
+
+
+class Upload(Service):
+    """A JSON service version of @@upload"""
+
+    def reply(self):
+        data = json_body(self.request)
+        surveygroup_title = data.get("surveygroup_title", "")
+        survey_title = data.get("survey_title", "OiRA Tool import")
+        upload = data.get("upload", None)
+        is_etranslate_compatible = data.get("is_etranslate_compatible", False)
+        if upload is None:
+            self.request.response.setStatus(400)
```

Comment:
I would move the check on top:
```suggestion
        upload = data.get("upload", None)
        if upload is None:
            self.request.response.setStatus(400)

        surveygroup_title = data.get("surveygroup_title", "")
        survey_title = data.get("survey_title", "OiRA Tool import")
        is_etranslate_compatible = data.get("is_etranslate_compatible", False)
```